src/Composer/Satis/Builder/ArchiveBuilder.php

```diff
@@ -99,9 +101,19 @@ public function dump(array $packages)
                 if (!$this->skipErrors) {
                     throw $exception;
                 }
+
+                //If file exists then exclude it
+                $packageName = $archiveManager->getPackageFilename($package);
+                if ('pear-library' === $package->getType()) {
+                    $excludeFiles[] = $packageName.'.'. pathinfo($package->getDistUrl(), PATHINFO_EXTENSION);
+                } else {
+                    $excludeFiles[] = $packageName . '.' . $format;
+                }
```

Comment:
@JamesRezo `$path = $archiveManager->archive($package, $format, $directory);` can throw Exception and `$path` will be:
- null
- `$path = realpath($directory).'/'.$packageName.'.'.pathinfo($package->getDistUrl(), PATHINFO_EXTENSION);`
- value set in the previous iteration